src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -167,4 +169,31 @@ public static DetailAST getAnnotation(final DetailAST ast,
         return result;
     }
 
+    /**
+     * Checks if the given AST element is marked with an annotation that allows
+     * it to omit javadoc documentation.
+     * @param ast The type or method definition.
+     * @param allowedAnnotations A collection of annotations that allow omitting documentation.
+     * @return true if the type or method is exempt from comments, false otherwise.
+     */
+    public static boolean hasAllowedAnnotations(DetailAST ast, List<String> allowedAnnotations) {
```

Comment:
Since this is a utility method, it should be named something common to everything it can be used for.
`hasAllowedAnnotations` => `hasAnnotation`.
`allowedAnnotations` => `list` (or if someone has a better name)

Reply:
fixed